pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -432,16 +477,21 @@ class ModelInterpreter implements Serializable {
         }
     }
 
+    @Deprecated
+    def inWrappers(Options options, Closure body) {
+        return inWrappers(options?.wrappers, body)
+    }
+
     /**
      * Executes the given closure inside 0 or more wrapper blocks if appropriate
-     * @param options The options configuration we're executing in
+     * @param wrappers A map of wrapper names to wrappers
      * @param body The closure to execute
      * @return The return of the resulting executed closure
      */
-    def inWrappers(Options options, Closure body) {
-        if (options?.wrappers != null) {
+    def inWrappers(Map<String,Object> wrappers, Closure body) {
```

Comment:
I don't see the reason for this change. You've just replaced the calls with `something.options.wrappers` instead.

Reply:
This may be an artifact of an earlier iteration where I wasn't supporting per-stage options...I'll revisit.

Reply:
Oh, no, duh on me - it's because `root.options` is `Options.class` while `stage.options` is `StageOptions.class` and I couldn't figure out a good way to make them the same class, so hey. =)

Reply:
There is this thing in Java called `interface` ;) although usually discouraged in Jenkins plugin development due to it being hard to handle binary compatibility going forward. But iirc we've used that quite extensively in this plugin :)